src/main/java/net/javadiscord/javabot/systems/user_commands/leaderboard/QOTWLeaderboardSubcommand.java

```diff
@@ -0,0 +1,205 @@
+	private static final int DISPLAY_COUNT = 10;
+	private static final int MARGIN = 40;
+
+	/**
+	 * The image's width.
+	 */
+	private static final int WIDTH = 3000;
+
+	public QOTWLeaderboardSubcommand() {
+		setSubcommandData(new SubcommandData("qotw", "The QOTW Points Leaderboard."));
+	}
+
+	@Override
+	public void execute(SlashCommandInteractionEvent event) {
+		event.deferReply().queue();
+		Bot.asyncPool.submit(() -> {
+			try {
+				QOTWPointsService service = new QOTWPointsService(Bot.dataSource);
+				WebhookMessageAction<Message> action = event.getHook().sendMessageEmbeds(buildLeaderboardRankEmbed(event.getMember(), service));
+				// check whether the image may already been cached
+				byte[] array = ImageCache.isCached(getCacheName()) ?
+						// retrieve the image from the cache
+						getOutputStreamFromImage(ImageCache.getCachedImage(getCacheName())).toByteArray() :
+						// generate an entirely new image
+						generateLeaderboard(event.getGuild(), service).toByteArray();
+				action.addFile(new ByteArrayInputStream(array), Instant.now().getEpochSecond() + ".png").queue();
+			} catch (IOException e) {
+				ExceptionLogger.capture(e, getClass().getSimpleName());
+			}
+		});
+	}
+
+	/**
+	 * Builds the Leaderboard Rank {@link MessageEmbed}.
+	 *
+	 * @param member  The member which executed the command.
+	 * @param service The {@link QOTWPointsService}.
+	 * @return A {@link MessageEmbed} object.
+	 */
+	private MessageEmbed buildLeaderboardRankEmbed(Member member, QOTWPointsService service) {
+		int rank = service.getQOTWRank(member.getIdLong());
+		String rankSuffix = switch (rank % 10) {
+			case 1 -> "st";
+			case 2 -> "nd";
+			case 3 -> "rd";
+			default -> "th";
+		};
+		long points = service.getPoints(member.getIdLong());
+		String pointsText = points == 1 ? "point" : "points";
+		return new EmbedBuilder()
+				.setAuthor(member.getUser().getAsTag(), null, member.getEffectiveAvatarUrl())
+				.setTitle("Question of the Week Leaderboard")
+				.setDescription(String.format("You're currently in `%s` place with `%s` %s.",
+						rank + rankSuffix, points, pointsText))
+				.setTimestamp(Instant.now())
+				.build();
+	}
+
+	/**
+	 * Draws a single "user card" at the given coordinates.
+	 *
+	 * @param g2d     Graphics object.
+	 * @param member  The member.
+	 * @param service The {@link QOTWPointsService}.
+	 * @param y       The y-position.
+	 * @param left    Whether the card should be drawn left or right.
+	 * @throws IOException If an error occurs.
+	 */
+	private void drawUserCard(Graphics2D g2d, Member member, QOTWPointsService service, int y, boolean left) throws IOException {
+		BufferedImage card = ImageGenerationUtils.getResourceImage("assets/images/LeaderboardUserCard.png");
+		int x = left ? MARGIN * 5 : WIDTH - (MARGIN * 5) - card.getWidth();
+		g2d.drawImage(ImageGenerationUtils.getImageFromUrl(member.getUser().getEffectiveAvatarUrl() + "?size=4096"), x + 185, y + 43, 200, 200, null);
```

Comment:
What if the user has a guild-specific avatar (Nitro) or no avatar set?

Reply:
Thats what `getEffectiveAvatarUrl` does iirc

Reply:
[`Member#getEffectiveAvatarUrl`](https://javadoc.io/doc/net.dv8tion/JDA/latest/net/dv8tion/jda/api/entities/Member.html) does but [`User#getEffectiveAvatarUrl`](https://javadoc.io/doc/net.dv8tion/JDA/latest/net/dv8tion/jda/api/entities/User.html) does **not**. Here, you are calling `Member#getEffectiveAvatarUrl` because of `Member#getUser`.